ckan/logic/__init__.py

```diff
@@ -329,7 +329,7 @@ def check_access(action: str,
     if not context.get('ignore_auth'):
         if not context.get('__auth_user_obj_checked'):
             if context["user"] and not context["auth_user_obj"]:
-                context['auth_user_obj'] = model.User.by_name(context['user'])
+                context['auth_user_obj'] = model.User.get(context['user'])
```

Comment:
This was discussed in https://github.com/ckan/ckan/pull/6762 and there were some concerns raised about it. Was this change required by Flask login?

Reply:
I believe I encountered an error while testing it because some context ['user'] contains user_id, like this [one](https://github.com/ckan/ckan/blob/96cb766218f723a398cd2a03059bd32b978701f1/ckan/views/user.py#L680-L688)

should I revert it and change that id to `current_user.name` instead?